databricks/koalas/groupby.py

```diff
@@ -1721,3 +1849,20 @@ def transform(self, func):
 
     def filter(self, func):
         raise NotImplementedError()
+
+    def rank(self, method='average', ascending=True):
+        kdf = super(SeriesGroupBy, self).rank(method, ascending).to_dataframe()
+        return _col(DataFrame(kdf._internal.copy(
+            sdf=kdf._sdf.select(kdf._internal.data_scols), index_map=[])))  # index is lost.
```

Comment:
Seems you do something with rank here. After https://github.com/databricks/koalas/commit/f2a718dcfdda965ac83128d8d979d3146e3ae76a, we should always set the index.

If index is lost, you can imply to:

```python
_col(DataFrame(kdf._internal.copy(sdf=kdf._sdf.select(kdf._internal.data_scols)))
```